### PR TITLE
add black formatting

### PR DESCRIPTION
**Pull request by karmacoma-eth**
bells and whistles:

- updated dev instructions in README.md
- pre-commit hook
- lint GitHub action 🧑‍⚖️
- mega refactor commit added to [.git-blame-ignore-revs](https://github.com/a16z/halmos/pull/119/commits/aecf3d8d040d33fad4bd978537df02e262b8b788)

### CODE REVIEW

File: tests/test_sevm.py [hunk 1]
```diff
@@ -68,7 +99,9 @@ def test_run(hexcode, stack, pc, opcode: int, sevm, solver, storage):
     assert ex.pc == pc
     assert ex.current_opcode() == int_of(opcode)
 
+
 def byte_of(i, x):
+    # fmt: off
```
Comment (karmacoma-eth) [inline tests/test_sevm.py:104]: this is the only customization I made

Comment (daejunpark) [reply tests/test_sevm.py:104]: let's not make an exception, as it is subjective to determine when it is acceptable.

there are other places where the formatted code is very ugly.  we can work on them in separate prs by rewriting them in a formatting-friendly way.

File: README.md [hunk 2]
```diff
@@ -79,6 +79,37 @@ Counterexample: [p_price_uint96 = 39614081294025656978550816768, p_quantity_uint
...
+# if you want to submit a pull request, fork the repository:
+
+gh repo fork a16z/halmos
+
+# or if you just want to develop locally, clone it
+# git clone git@github.com:a16z/halmos.git
+
+# create and activate a virtual environment
+
+python3.10 -m venv .venv
```
Comment (daejunpark) [inline README.md:94]: 3.10 doesn't seem representative.  we support >=3.8 and the latest is 3.11.  can we just say `python3`?

Comment (karmacoma-eth) [reply README.md:94]: let's recommend 3.11 then, makes more sense to me

File: tests/test_sevm.py (hunk 1 above)
Comment (daejunpark) [reply tests/test_sevm.py:104]: ah, i will take back. this is a test file, and i couldn't think of a good way to write it in the format friendly way (it needs to be verbose to cross-check with the main code logic.)

Comment (karmacoma-eth) [reply tests/test_sevm.py:104]: image

**Review by daejunpark — APPROVED** [review]
looks great, thanks!

File: tests/test_sevm.py (hunk 1 above)
Comment (karmacoma-eth) [reply tests/test_sevm.py:104]: yeah I was going to say I agree about avoiding exceptions, but given that this is a test file and the extreme nesting makes black spiral into madness, so I think an exception is fine 

image

**Review by daejunpark — APPROVED** [review]

**Pull request merged**